src/components/FormDatePicker.vue

```diff
@@ -76,20 +76,27 @@
       dataFormat: {
         immediate: true,
         handler() {
-          this.config.format = this.dataFormat === 'datetime' ? this.getUserDateTimeFormat() : this.getUserDateFormat();
-          this.date = this.stdValue(this.value);
+          this.date = this.pickerValue(this.value);
         }
       },
-      value(value) {
-        this.date = this.stdValue(value);
+      value: {
+        immediate: true,
+        handler(value) {
+          this.date = this.pickerValue(value);
+        }
       },
-      date() {
+      date(date) {
         if (typeof this.date === "string") {
-          this.stdValue(this.value) !== this.stdValue(this.date) ? this.$emit('input', this.stdValue(this.date)) : null;
+          date = moment(date, this.config.format).format(datetimeStdFormat);
```

Comment:
Should this line not use `stdValue`? It seems like it will format the date to `datetimeStdFormat` regardless of whether the user chose date or datetime for their format. And to add to the confusion, the watcher for `value` above, and the `mounted` method, both set the date using `pickerValue`, which yet again uses a different format!

I think these methods need to be refactored to either avoid some redundant formatting, or to make the intentions easier to understand.